### PR TITLE
Add categorical support for pandas and dask backend

### PR DESCRIPTION
**Pull request by jcrist**
Can now discover pandas dataframes with categorical dtype. Also implemented discover for ``dask.dataframe.DataFrame`` and ``dask.dataframe.Series``.

### CODE REVIEW

**Comment by kwmsmith** [comment]
One request, please make a `whatsnew/0.5.1.txt` document and add a bullet point for this PR.

Overall, LGTM.

@llllllllll you've mentioned categorical support in the past, I believe -- can you take a look?

File: odo/backends/dask.py [hunk 1]
```diff
@@ -22,6 +21,12 @@ def discover_dask_array(a, **kwargs):
     return from_numpy(a.shape, a.dtype)
 
 
+@discover.register(dd.Series)
+@discover.register(dd.DataFrame)
+def discover_dask_dataframe(df):
+    return var * discover(df.head()).measure
```
Comment (llllllllll) [inline odo/backends/dask.py]: is `len` a potentially expensive computation with a dask structure?

Comment (jcrist) [inline odo/backends/dask.py]: Yeah, it requires a full graph computation (dataframes don't store their size as part of the metadata).

Comment (llllllllll) [inline odo/backends/dask.py]: Just making sure, thanks

File: odo/backends/tests/test_dask_dataframe.py [hunk 2]
```diff
@@ -8,10 +8,24 @@
...
+
+
+def test_discover():
+    df = pd.DataFrame({'x': list('a'*5 + 'b'*5 + 'c'*5),
+                       'y': range(15),
+                       'z': list(map(float, range(15)))},
+                       columns=['x', 'y', 'z'])
+    df.x = df.x.astype('category')
+    ddf = dd.from_pandas(df, npartitions=2)
+    assert discover(ddf) == ds.var * ds.Record([('x', ds.Categorical(['a', 'b', 'c'])),
```
Comment (llllllllll) [inline odo/backends/tests/test_dask_dataframe.py]: could you use `datashape.util.assert_dshape_equal` to generate better error messages in the case where these are not equal?

**Comment by llllllllll** [comment]
Just the one comment about the tests. Looks good, thank you

**Comment by jcrist** [comment]
Fixed tests and added whatsnew page. Good to merge?

**Comment by llllllllll** [comment]
Yup

**Pull request merged**